src/mqueryfront/src/QueryPage.js

```diff
@@ -77,127 +98,115 @@ class QueryPage extends Component {
         return [...new Set(taintList)];
     }
 
-    updateQhash(newQhash, rawYara) {
-        if (typeof rawYara !== "undefined") {
-            this.setState({ rawYara: rawYara });
-        }
+    updateYara(value) {
+        this.setState({ rawYara: value });
+    }
 
-        let path;
-        if (!newQhash) {
-            path = "/";
+    async trackJob() {
+        // Periodically reloads job status until job is finished
+        let { job, matches } = await this.loadMatches();
+
+        this.setState({ job, matches });
+        if (!isStatusFinished(job.status)) {
+            this.trackJobTimeout = setTimeout(() => this.trackJob(), 1000);
         } else {
-            path = "/query/" + newQhash;
-            this.collapsePane();
+            this.trackJobTimeout = null;
         }
-        this.props.history.push(path, { internal: true });
+    }
 
-        this.setState({
-            mode: "job",
-            queryError: null,
-            queryPlan: null,
-            qhash: newQhash,
-            matches: null,
-            job: null,
-            activePage: 1,
-        });
-        this.loadJob();
+    cancelJob() {
+        // Cancels perodic job status reload
+        if (this.trackJobTimeout !== null) {
+            clearTimeout(this.trackJobTimeout);
+            this.trackJobTimeout = null;
+        }
     }
 
-    updateYara(value) {
-        this.setState({ rawYara: value });
+    setActivePage(pageNumber) {
+        this.setState({ activePage: pageNumber }, async () => {
+            this.setState(await this.loadMatches());
+        });
     }
 
-    loadJob() {
+    async loadMatches() {
+        // Loads matches from the current page
         const LIMIT = 20;
-        let OFFSET = (this.state.activePage - 1) * 20;
-
-        if (!this.state.qhash) {
-            return;
+        const OFFSET = (this.state.activePage - 1) * LIMIT;
+        const response = await axios.get(
+            API_URL +
+                "/matches/" +
+                this.queryHash +
+                "?offset=" +
+                OFFSET +
+                "&limit=" +
+                LIMIT
+        );
+        if (response) {
+            const { job, matches } = response.data;
+            return { job, matches };
```

Comment:
is it different than `return response.data`?

Reply:
🤔 Yup, this looks like unnecessary filter for these two keys..